commons/protocol/src/main/scala/sbt/protocol/Keys.scala

```diff
@@ -98,3 +98,37 @@ object MinimalBuildStructure {
   implicit val unpickler: Unpickler[MinimalBuildStructure] = genUnpickler[MinimalBuildStructure]
   implicit val pickler: Pickler[MinimalBuildStructure] = genPickler[MinimalBuildStructure]
 }
+
+final case class Attributed[T](data: T)
+object Attributed {
+
+  implicit def unpickler[T: Unpickler]: Unpickler[Attributed[T]] = new Unpickler[Attributed[T]] {
+
+    override val tag = implicitly[FastTypeTag[Attributed[T]]]
+
+    override def unpickle(tag: String, reader: PReader): Any = {
+      reader.hintTag(pickling.FastTypeTag.String)
```

Comment:
though I'm not sure why these hints are needed at all, maybe it's because in the pickler you need to write out the tag